src/Symfony/Component/HttpClient/RetryableHttpClient.php

```diff
@@ -120,6 +120,7 @@ public function request(string $method, string $url, array $options = []): Respo
             ]);
 
             $context->replaceRequest($method, $url, $options);
+            $context->setInfo('retry_count', $retryCount);
```

Comment:
why this line? isn't the increased value already updated on the next hop with the same lines above? if needed, shouldn't it be setonly when `$retryCount < $this->maxRetries`?

Reply:
Indeed, the value will be set on the next hop, 

I moved to `if ($retryCount >= $this->maxRetries) {` bloc. Issue is there is no hop for last retry.